x-pack/plugins/apm/public/components/app/RumDashboard/URLFilter/URLSearch/index.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSelectableTemplateSitewide,
+  EuiSelectableTemplateSitewideOption,
+  EuiTitle,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import useDebounce from 'react-use/lib/useDebounce';
+import React, { useEffect, useState, FormEvent, useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useUrlParams } from '../../../../../hooks/useUrlParams';
+import { useFetcher } from '../../../../../hooks/useFetcher';
+import { I18LABELS } from '../../translations';
+import { fromQuery, toQuery } from '../../../../shared/Links/url_helpers';
+import { formatToSec } from '../../UXMetrics/KeyUXMetrics';
+
+interface Props {
+  onChange: (value: string[]) => void;
+}
+
+export function URLSearch({ onChange: onFilterChange }: Props) {
+  const history = useHistory();
+
+  const { urlParams, uiFilters } = useUrlParams();
+
+  const { start, end, serviceName } = urlParams;
+  const [searchValue, setSearchValue] = useState('');
+
+  const [forceClosePopover, setForceClosePopover] = useState(false);
+
+  const [debouncedValue, setDebouncedValue] = useState('');
+
+  const [searchRef, setSearchRef] = useState<HTMLInputElement | null>(null);
+
+  const onInputFocus = () => {
+    setForceClosePopover(false);
+  };
+
+  useEffect(() => {
+    if (searchRef) {
+      searchRef.addEventListener('focus', onInputFocus);
+    }
+    return () => {
+      searchRef?.removeEventListener('focus', onInputFocus);
+    };
+  }, [searchRef]);
+
+  useDebounce(
+    () => {
+      setSearchValue(debouncedValue);
+    },
+    250,
```

Comment:
We might want to slow this down a bit more. I am a fast typist and it works great for me, but if someone is typing slowly while thinking, or just a slower typist, it might be better to have a slower value like `500`.

Reply:
i copied kibana nav search timeout, 500 adds some user noticable delay